package-scripts.js

```diff
@@ -0,0 +1,106 @@
+        },
+        requires: test('requires', '--compilers coffee:coffee-script/register ' +
+          '--require test/require/a.js ' +
+          '--require test/require/b.coffee ' +
+          '--require test/require/c.js ' +
+          '--require test/require/d.coffee ' +
+          'test/require/require.spec.js'),
+        reporters: test('reporters', 'test/reporters/*.spec.js'),
+        only: {
+          default: 'nps test.node.only.bdd test.node.only.tdd test.node.only.bddRequire',
+          bdd: test('only-bdd', '--ui bdd test/only/bdd.spec'),
+          tdd: test('only-tdd', '--ui tdd test/only/tdd.spec'),
+          bddRequire: test('only-bdd-require', '--ui qunit test/only/bdd-require.spec')
+        },
+        globalOnly: {
+          default: 'nps test.node.globalOnly.bdd test.node.globalOnly.tdd test.node.globalOnly.qunit',
+          bdd: test('global-only-bdd', '--ui bdd test/only/global/bdd.spec'),
+          tdd: test('global-only-tdd', '--ui tdd test/only/global/tdd.spec'),
+          qunit: test('global-only-qunit', '--ui qunit test/only/global/qunit.spec')
+        }
+      },
+      browser: {
+        default: 'nps ' +
+          'clean ' +
+          'build.mochajs ' +
+          'test.browser.unit ' +
+          'test.browser.bdd ' +
+          'test.browser.tdd ' +
+          'test.browser.qunit ' +
+          'test.browser.esm',
+        unit: `NODE_PATH=. karma start --single-run`,
+        bdd: 'MOCHA_TEST=bdd nps test.browser.unit',
+        tdd: 'MOCHA_TEST=tdd nps test.browser.unit',
+        qunit: 'MOCHA_TEST=qunit nps test.browser.unit',
+        esm: 'MOCHA_TEST=esm nps test.browser.unit'
+      },
+      nonTTY: {
+        default: 'nps test.nonTTY.dot test.nonTTY.list test.nonTTY.spec',
+        dot: test('non-tty-dot', '--reporter dot test/interfaces/bdd.spec 2>&1 > /tmp/dot.out && echo "dot:" && cat /tmp/dot.out'),
+        list: test('non-tty-list', '--reporter list test/interfaces/bdd.spec 2>&1 > /tmp/list.out && echo "list:" && cat /tmp/list.out'),
+        spec: test('non-tty-dot', '--reporter spec test/interfaces/bdd.spec 2>&1 > /tmp/spec.out && echo "spec:" && cat /tmp/spec.out')
+      }
+    },
+    tm: `open editors/${TM_BUNDLE}`,
+    prepublishOnly: 'nps test clean build',
+    coveralls: 'nyc report --reporter=text-lcov | coveralls',
+    prebuildDocs: 'rm -rf docs/_dist && node scripts/docs-update-toc.js',
+    buildDocs: 'nps prebuildDocs && bundle exec jekyll build --source ./docs --destination ./docs/_site --config ./docs/_config.yml --safe --drafts && nps postbuildDocs',
+    postbuildDocs: 'buildProduction docs/_site/index.html --outroot docs/_dist --canonicalroot https://mochajs.org/ --optimizeimages --svgo --inlinehtmlimage 9400 --inlinehtmlscript 0 --asyncscripts && cp docs/_headers docs/_dist/_headers && node scripts/netlify-headers.js >> docs/_dist/_headers',
+    prewatchDocs: 'node scripts/docs-update-toc.js',
```

Comment:
I can't remember if `nps` supports `pre-` and `post-` scripts.  does it?  when we were running this with `npm`, `prewatchDocs` would get run before `watchDocs` when doing `npm run watchDocs`

Reply:
Don't think it does, but the `pre` and `post` commands are explicitly inserted into the nps commands for that. I did move `prepublishOnly` back to `package.json` as I suspect it wouldn't trigger on `npm publish` if it was in `package-scripts.js`